cli/pub_command.go

```diff
@@ -146,16 +151,43 @@ func (c *pubCmd) doReq(nc *nats.Conn, progress *uiprogress.Bar) error {
 			return err
 		}
 
-		m, err := nc.RequestMsg(msg, opts.Timeout)
+		msg.Reply = nc.NewRespInbox()
+
+		s, err := nc.SubscribeSync(msg.Reply)
 		if err != nil {
 			return err
 		}
 
-		if c.raw {
-			fmt.Println(string(m.Data))
-		} else if progress != nil {
-			progress.Incr()
-		} else {
+		err = nc.PublishMsg(msg)
+		if err != nil {
+			return err
+		}
+
+		// loop through the reply count.
+		start := time.Now()
+
+		// Honor the overall timeout for the first response.  No
+		// responders will circuit break.
+		timeout := opts.Timeout
+
+		// loop until reply count is met, or if zero, until we
+		// timeout receiving messages.
+		rc := 0
+		for {
+			m, err := s.NextMsg(timeout)
+			if err != nil {
+				if err == nats.ErrTimeout {
+					return nil
+				}
+				if err == nats.ErrNoResponders {
+					log.Printf("No responders are available")
+					return nil
+				}
+
+				log.Printf("Error receiving message: %v", err)
+				return err
```

Comment:
this will log here and in the exit handler wont it? Error would be logged twice? maybe better `return fmt.Error("could not receive message: %v", err)` which would then be handled and logged higher up